src/core/audio/AudioJack.cpp

```diff
@@ -203,51 +183,43 @@ bool AudioJack::initJackClient()
 
 void AudioJack::startProcessing()
 {
-	if( m_active || m_client == nullptr )
+	if (m_active || m_client == nullptr)
 	{
 		m_stopped = false;
 		return;
 	}
 
-	if( jack_activate( m_client ) )
+	if (jack_activate(m_client))
 	{
-		printf( "cannot activate client\n" );
+		printf("cannot activate client\n");
 		return;
 	}
 
 	m_active = true;
 
-
 	// try to sync JACK's and LMMS's buffer-size
-//	jack_set_buffer_size( m_client, audioEngine()->framesPerPeriod() );
+	//	jack_set_buffer_size( m_client, audioEngine()->framesPerPeriod() );
 
-
-
-	const char * * ports = jack_get_ports( m_client, nullptr, nullptr,
-						JackPortIsPhysical |
-						JackPortIsInput );
-	if( ports == nullptr )
+	const char** ports = jack_get_ports(m_client, nullptr, nullptr, JackPortIsPhysical | JackPortIsInput);
+	if (ports == nullptr)
 	{
-		printf( "no physical playback ports. you'll have to do "
-			"connections at your own!\n" );
+		printf("no physical playback ports. you'll have to do "
+			   "connections at your own!\n");
 	}
 	else
 	{
-		for( ch_cnt_t ch = 0; ch < channels(); ++ch )
+		for (ch_cnt_t ch = 0; ch < channels(); ++ch)
 		{
-			if( jack_connect( m_client, jack_port_name(
-							m_outputPorts[ch] ),
-								ports[ch] ) )
+			if (jack_connect(m_client, jack_port_name(m_outputPorts[ch]), ports[ch]))
 			{
-				printf( "cannot connect output ports. you'll "
-					"have to do connections at your own!\n"
-									);
+				printf("cannot connect output ports. you'll "
+					   "have to do connections at your own!\n");
 			}
 		}
 	}
 
 	m_stopped = false;
-	free( ports );
+	free(ports);
```

Comment:
According to the [API docs](https://jackaudio.org/api/group__PortSearching.html), we should be calling `jack_free` here.
See also: https://jackaudio.org/api/group__TimeFunctions.html#ga437010097629ac8124a56c752c02f299

Reply:
Thanks, good find. I fixed that.

Reply:
From the logs included in issue #5954, LMMS printed _"cannot connect output ports. you'll have to do connections at your own!"_ just before crashing, and looking at the code, that is the message it prints right before it called `free(ports)`.

And according the the Jack docs, failure to call `jack_free` can cause crashes on Windows:
> This is MANDATORY on Windows when otherwise all nasty runtime version related crashes can occur.

So it seems likely to me that calling `jack_free` may have fixed #5954.